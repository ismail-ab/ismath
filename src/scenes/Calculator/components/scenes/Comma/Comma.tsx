import * as React from 'react';
import { CharacterButton } from '../../components';

const Comma: React.SFC = () => (
  <CharacterButton
    characterToDisplay="."
    color="primary"
  />
);

export default Comma;