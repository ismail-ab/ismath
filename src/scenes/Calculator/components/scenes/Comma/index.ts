export { default as Comma } from './Comma';