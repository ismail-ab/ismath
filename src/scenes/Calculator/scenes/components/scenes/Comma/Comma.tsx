import * as React from 'react';
import { CharacterButton } from '../../components';
import { CalculatorContext } from '../../../contexts';

const Comma: React.SFC = () => (
  <CalculatorContext.Consumer>
    {({ concatenateCharacters }) => (
      <CharacterButton
        characterToDisplay="."
        color="primary"
      />
    )}
  </CalculatorContext.Consumer>
);

export default Comma;