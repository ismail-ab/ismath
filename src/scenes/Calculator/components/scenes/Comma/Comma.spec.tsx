import { } from 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Comma from './Comma';

describe('Comma', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Comma />);

    expect(wrapper).toMatchSnapshot();
  });
});
