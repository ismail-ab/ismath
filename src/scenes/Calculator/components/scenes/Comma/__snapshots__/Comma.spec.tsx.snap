// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comma renders correctly 1`] = `
<CharacterButton
  characterToDisplay="."
  color="primary"
/>
`;
